### ISSUE TITLE
Beta Testing: X360CE 4.8.x.x Alpha

### ISSUE THREAD

**Issue by EJocys** [x360ce/x360ce#818]
This issue is for reporting bugs found in Beta Testing: X360CE 4.8.x.x Alpha. 

This release contains better exception logging. Removed more old GDB/INI/DLL code. Start with Windows and support for Profiles added. Rest of the focus was on cloud database and multiple profiles support. Default build is AnyCPU now (one x360ce.exe for 32-bit and 64-bit OS). Exclusive 32-bit and 64-bit are still available and I can supply them in the future if there will be problems with MSIL (AnyCPU) builds. Other fixes are listed on release page.

Alpha is intended to test Virtual Emulation. Download links and instructions can be found here:
https://github.com/x360ce/x360ce/blob/master/Wiki/BetaTesting.md

More details about last release here:
https://github.com/x360ce/x360ce/releases

**Comment by KiNGKiMO** [comment]
Thank you @EJocys,

x360ce crashes at startup:

DllNotFoundException: Unable to load DLL 'vigemclient.dll': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

Edit:
Downloaded the missing dll from "x360ce.App.Beta/Resources/ViGEmClient/x64" and it started, but I never had to use vigemclient.dll before.

Edit 2:
I've now noticed that x360ce creates "ProgramData\X360CE\Temp\ViGEmClient.dll.84A31178" folder with vigemclient.dll, but this is the x86 version, I replaced with the x64 version I downloaded and it worked... so the correct ViGEmClient.dll needs to be extracted depending on OS architecture I suppose.

**Comment by KiNGKiMO** [comment]
Can we add Start Minimized? or maybe use the same option selected at Start with Windows?

**Comment by EJocys** [comment]
@KiNGKiMO: Thanks for reporting. Fixed VigemClient.dll issue. X360CE.exe process is using it, so it extracts correct one now (Depending on which architecture, 32-bit or 64-bit, x360ce application runs at the moment).

Yes, you can start minimised. There are two options to do that. You can:

a) Start executable with "WindowState"option. Example: x360ce.exe /WindowState=Minimized
b) Create shortcut (for example on the Desktop). Open shortcut properties via right-mouse click and set "_Run_:" option to _Minimised_.

You can re-donwload latest update from here: https://github.com/x360ce/x360ce/releases
Edit: Re-uploaded release zips with correct 4.8.0.14 version.

**Comment by KiNGKiMO** [comment]
@EJocys Thank you.
Updated release works just fine.

Do we still need all xml files at "ProgramData\X360CE" ?

**Comment by EJocys** [comment]
Only these XMLs are used.  They will be recreated if deleted:
x360ce.Layouts.xml
x360ce.Options.xml
x360ce.PadSettings.xml
x360ce.Presets.xml
x360ce.Programs.xml
x360ce.Summaries.xml
x360ce.UserDevices.xml
x360ce.UserGames.xml
x360ce.UserInstances.xml
x360ce.UserSettings.xml
x360ce.XInputMask.xml - This is no longer needed. I will remove it in next release.

Note: Each file have XML comment inside at the top, which describes its purpose. Some of files will be larger later, because most popular world settings will be stored in them, so that x360ce application can set many controllers correctly automatically without connecting to the Internet.

**Comment by Squall-Leonhart** [comment]
So im not getting the stuck vibration when starting x360ce anymore, but I'm not getting directional constant vibration with the constant setting either.

Swap motors is either both small or both large instead of swapping the 2.

Edit
And now its working after toggling "Enable force feedback"

So it seems FFB is still not initializing properly at a fresh start

**Comment by Squall-Leonhart** [comment]
Looks like the reason is this anycpu stuff,  i'd rather have a 32bit only version :\

many controls just don't have a 64bit FFB library

**Comment by Squall-Leonhart** [comment]
>Fix force feedback not working when controller re-connected.

This is still not working in 4.8.0.14

**Comment by EJocys** [comment]
@Squall-Leonhart Maybe, I fixed just part of the problem. I will try to replicate the issue again.

**Comment by KiNGKiMO** [comment]
@EJocys 
`Enabling Virtual Drivers:

[Issues] tab will start blinking it Virtual Drivers are missing.
Press [Issues] tab and hit [Install] button to install Virtual Controller Drivers.`

This is not happening, I've tried it uninstalling ViGEM Bus and nothing at [Issues] tab... Please recheck.

Thanks.

**Comment by Squall-Leonhart** [comment]
there are updates to vigembus for a couple of bsod issues that we should take in, and hidguardian has been overhauled.

**Comment by KiNGKiMO** [comment]
@Squall-Leonhart 
This explains the BSOD I get every now and then.

New versions of hidguardian are totally different as i recall. Will need to be re-implemented in x360ce to work correctly.

**Comment by Squall-Leonhart** [comment]
it might, but it also might not.

**Comment by KiNGKiMO** [comment]
@EJocys Any new alphas after hidguardian update? I didn't know there's a 1.14.3.0 version.

**Comment by KiNGKiMO** [comment]
I tried x360ce today on an old machine (Core 2 Quad Processor Q9650/8GB RAM) Fresh installed Windows 10 ltsc 2019.
I tested Pro Evolution Soccer 2018 and the frame rate drops from 60 to 5x while using x360ce making stuttering effects at the game.

The game works just fine with x360ce closed though.

This is the 1st time I'm facing such issue... Any ideas?

Edit:
Well, I've discovered that it has nothing to do with x360ce... It's the game itself, it works just fine if the Stadium settings were set to "Night"... it was random before, so when it's "Day" the game stutters on older CPUs.

**Comment by tmarquesbr** [comment]
@KiNGKiMO I saw that once with some games and I got help. I tried to turn off "Enable Force Feedback" and that worked for me. The motor won't work but...

Also, I found two issues using a pair of controllers here. I don't know if that can count as a bug (and I didn't find any solution here), but...

Controllers used for this: Ipega PG-9076 (bluetooth, wired and USB dongle) and a B-MAX (wired)

1) Turning off Ipega using bluetooth mode (or letting it to turn off by itself) is forcing X360ce to close without messages. I took the file at the errors' folder. It's small but it's this...

NullReferenceException: ReferÃªncia de objeto nÃ£o definida para uma instÃ¢ncia de um objeto.System.NullReferenceException: Referência de objeto não definida para uma instância de um objeto.
   at x360ce.Engine.Data.UserDevice.LoadHidDeviceInfo(DeviceInfo DeviceInfo)

The controller works wired and with a USB dongle, which is a workaround for me. I don't have another bluetooth controller to check for more issues (sorry). Tried to turn off any HiD but not only failed but the controller fails to connect with x360ce.

2) When there's a combined mode (considering bluetooth disconnected since it'll generate issue 1), turning off one of them will force all Axis and POVs to position 0 (upwards) until you turn that specific controller on again (and with Ipega, using the mode I was using prior to the issue). That, of course, affects all other controllers connected to it. I tried to find a possible workaround without restarting the program but no idea to figure it out.

**Comment by KiNGKiMO** [comment]
@tmarquesbr 
Thanks, I edited the post, its the game itself.

Regarding your issue, it happens with me too, sometimes x360ce crashes when removing connected devices, doesn't happen all the time though.

About 2. it happens even with joy.cpl, if you remove a connected device sometimes you'll see it's stuck to upper left corner.

**Comment by KiNGKiMO** [comment]
x360ce crashes when hiding device at "Map Device To Controller" Window.
![image](https://user-images.githubusercontent.com/11393200/50374553-679ecf00-05f8-11e9-9066-b28e5f59f721.png)







[Errors.zip](https://github.com/x360ce/x360ce/files/2705194/Errors.zip)

**Comment by KiNGKiMO** [comment]
@EJocys Hi, how are you?
Any plans for new alpha, or is it even ready for beta?

**Comment by Squall-Leonhart** [comment]
i think we were waiting on HID Guardian's rewrite before progressing further.

**Comment by KiNGKiMO** [comment]
Well, I don't think this is going to be any time soon (New HID Guardian I mean).
Current alpha is magnificent BTW, we just need fixing of those random crashes, mostly when devices disconnect/reconnect.
And then it's time moving toward x360ce service maybe, dont know your plan guys, also your time/life.

Appreciating your work always.

**Comment by EJocys** [comment]
I will integrate new version as soon as signed download and simple documentation, on how to integrate new HID Guardian/ViGEm, is available. Last HID Guardian/ViGEm release I've tried was too overcomplicated. Nefarius probably have some other priorities in his life at the moment. I am updating and fixing parts which are shared with other projects. Probably need to release new x360ce version, just to flush bugfixes and stability changes.

**Comment by KiNGKiMO** [comment]
True, especially regarding HID Guardian, a new version might not come in the near future if ever, but a new ViGEMBus with simple exe installer is on the way, it's almost ready says Nefarius.

Totally agree with releasing x360ce bugfixes and stability changes, this is what I meant actually.

**Comment by Squall-Leonhart** [comment]
@EJocys  we should pull in vigem 1.16.112 asap, as it resolves a rare bsod.

**Comment by KiNGKiMO** [comment]
Yep, 1.16.112 is released an hour ago.

**Comment by EJocys** [comment]
Good. Looks like new version works. I think, I will remove internal ViGEm installer from X360CE and replace [Install] [Uninstall] buttons with [Download Installer...], which redirects users to https://vigem.org.

**Comment by KiNGKiMO** [comment]
Well, I think the built in ViGEm installer with batch file is a better way, but if you like to change that, redirect the users to the github releases link:
https://github.com/ViGEm/ViGEmBus/releases

**Comment by KiNGKiMO** [comment]
I wanted to test force feedback now, I connected the TWIN USB Gamepad Adapter (PSX/PS2 to USB) while x360ce was running then I enabled the two controllers, and few seconds later x360ce crashed:

FCE_InvalidOperationException_20190421_212006.694074.htm
FCE_InvalidOperationException_20190421_212006.764392.htm

These are at Errors folder.
[Errors.zip](https://github.com/x360ce/x360ce/files/3101519/Errors.zip)

I then restarted x360ce and enabled the two controllers again and tested the force feedback, only the left motor works although both works correctly while testing dinput at joy.cpl.

**Comment by Squall-Leonhart** [comment]
I don't have much hope for that gamepads vibration driver, i don't think it even has the 64bit FF library implemented well.

my xbcdr modification has both motors run idividually and works with x360ce.

**Comment by KiNGKiMO** [comment]
Well, I thought same too... but few year ago I finally could find a driver that worked.

Here it is, since it's almost impossible to find:
[VID_0810&PID_0001 Twin USB Gamepad 64Bit Setup2.zip](https://github.com/x360ce/x360ce/files/3101905/VID_0810.PID_0001.Twin.USB.Gamepad.64Bit.Setup2.zip)

**Comment by KiNGKiMO** [comment]
@EJocys, thank you so much for the update.
1st thing I noticed is old settings not working, I had to reconfigure.

Check for updates on startup check box isn't saved, it's always selected when you restart x360ce.

Also, looks like we've an older version of ViGEmClient.dll.

EDIT:
Well, I wanted to compare old settings and new ones to know why exactly old settings did not work with alpha 4.9.0.0, so I noticed settings are created in a Settings folder now, that's why they did not work, they were at the root before.

**Comment by KiNGKiMO** [comment]
I've came across a very strange issue today, I tried a steam game called "Riverbond", if more than 1 controller enabled at x360ce none of them work at the game... You've to enable only 1 controller so you can use gamepad in-game, then you can alt+tab to x360ce and enable others controllers and they'll work just fine.

**Comment by Rogus** [comment]
Hey guys, any of you can help me how to make x360ce 4.10.0.0 Alpha build to work on steam? My controller is a wireless controller dualshock 4 I've tried to follow one guide on the community labeled as "How to make arcade fight sticks and dualshock 4 controllers or any controller to work" and in the guide the guy said to put x360ce the old version 3.2.82.0 version that re-release of old version on dragon ball fighterZ and I've always remembered that Denuvo Protected games aren't supported so, how did it worked on his side I've tried too putting alpha version but it gets double inputs. I have followed this tutorial: https://docs.google.com/document/d/19nLqBsll4ujRMI8rIfkZ-kIGqQ-U6Ns9UiqprA4o--A/edit# to no avail. So my issue with both alpha and old version it's that doesn't works.

**Comment by Squall-Leonhart** [comment]
the dual shock controller already has steam compatibility, it does not require x360ce.

**Comment by KiNGKiMO** [comment]
@Rogus 
You should hide the original controllers so you don't get double inputs...
1st install HID Guardian (at options tab).
2nd hide the original controllers (from devices tab).

And for dualshock 4, I recommended using DS4Windows by Ryochan7.

**Comment by Rogus** [comment]
@KiNGKiMO Thank you King for the help and @Squall-Leonhart Thanks man but in dragon ball fighterZ they have a problem with dinput also with Oninaki like @EJocys said for those Unity games they have a problem for controllers not being xinput. Thank you so much I will try your recommendations. Love all of your work on this software it's been helping me so much.

**Comment by EvoJocys** [comment]
This was the 4.8 alpha test thread. The 4.x line has since shipped as 4.17 through 4.21.

Closing as part of a tracker clean-up after the 4.21 release.
